drop-boxes/register-omero-metadata/register-omero.py

```diff
@@ -144,11 +157,75 @@ def findMetaDataFile(incomingPath):
 		for f in files:
 			stem, ext = os.path.splitext(f)
 			if ext.lower()=='.tsv':
-				with open(os.path.join(root, f), 'U') as fh: metadataFile = fh.readlines()
+				with open(os.path.join(root, f), 'U') as fh: metadataFileContent = fh.readlines()
 	return metadataFileContent
 
+def getPropertyNames(metadataFile):
+	"""Here we could add more complex behaviour later on.
+	"""
+	
+	property_names = metadataFile[0].split("\t")
+	for i in range(len(property_names)):
+		property_names[i] = property_names[i].strip().upper()
+
+	return property_names
+
+def validatePropertyNames(property_names):
+	"""Validate metadata property names.
+	TODO: call the imaging metadata parser (with json schema).
+	"""
+
+	# fast validation without parser object.
+	required_names = ["IMAGE_FILE_NAME", "IMAGING_MODALITY", "IMAGED_TISSUE", "INSTRUMENT_MANUFACTURER", "INSTRUMENT_USER", "IMAGING_DATE"]
+
+	for name in required_names:
+		if not name in property_names:
+			return False
+
+	return True
+
+def getPropertyMap(line, property_names):
+	"""Build the property map. Here we could add more complex behaviour later on.
+	"""
+
+	properties = {}
+	property_values = line.split("\t")
+
+	for i in range(1, len(property_names)): #exclude first col (filename)
+		##remove trailing newline, and replace space with underscore
+		name = property_names[i].rstrip('\n').replace(" ", "_")
+		value = property_values[i].rstrip('\n').replace(" ", "_")
+
+		properties[name] = value
+
+	return properties
+
+def filterOmeroPropertyMap(property_map):
+	"""Filters map before ingestion into omero server
+	"""
+
+	#the blacklist, e.g. what is going to openBIS or is automatically added to omero (e.g. file name)
+	filter_list = ["IMAGE_FILE_NAME", "INSTRUMENT_USER", "IMAGING_DATE"]
+
+	new_props = {}
+	for key in property_map.keys():
+		if not key in filter_list:
+			new_props[key] = property_map[key]
+
+	return new_props
+
```

Comment:
I think it is more declarative to pass the filter as argument, than secretly filter for sth not visible to the caller.
```suggestion
def filterOmeroPropertyMap(property_map, filter_list):
	"""Filters map before ingestion into omero server
	"""

	#the blacklist, e.g. what is going to openBIS or is automatically added to omero (e.g. file name)
	filter_list = ["IMAGE_FILE_NAME", "INSTRUMENT_USER", "IMAGING_DATE"]

	new_props = {}
	for key in property_map.keys():
		if not key in filter_list:
			new_props[key] = property_map[key]

	return new_props

```

Reply:
I added the filter "filter_list" as a parameter, but left the definition in the method, because this is also a placeholder for the actual metadata parser object (with the json schema), since the parser object should take care of filtering the properties imo

Reply:
thank you for you comment @luiskuhn. First i think you should expect developer to know what might happen in the future. Currently, you set a preset filter, and i think it is better to define that in the top level of the script and parse it as parameter to the function for now.
Second is the filtering in the parser. I think this violates the SRP. Let the parser be dumb. You can control on the schema, if you expect a fixed set of entries with values if you set them as required.

Reply:
I think we have different ideas of the tasks the metadata validation object should perform, but even if the validation/parser object to not know about it, you still need at some point functionality to separate the set of properties that go into openBIS and what goes in OMERO. This can be modeled with a different object, but what I mean is that `filterOmeroPropertyMap()` is a placeholder for more robust mapping functionality.
but I can define the filter list in the top-level and pass it as a parameter.

Reply:
done